### ISSUE TITLE
spot pricing not showing

### ISSUE THREAD

**Issue by chanpit-rakbank** [awslabs/eks-node-viewer#187]
Hi Team,

EKS node viewer showing only on-demand pricing , but it is not showing spot pricing. The spot instance created by karpenter and I can able to see label  "karpenter.sh/capacity-type | spot" in spot node.

![image](https://github.com/awslabs/eks-node-viewer/assets/144238035/f628f4e1-496f-400c-a050-8ae958a6f5cb)

**Comment by fnicolelli-ls** [comment]
Hi @chanpit-rakbank I had the same issue and I solved it by adding ec2:DescribeSpotPriceHistory in the iam policy.

**Comment by Tr3V3rn** [comment]
> Hi Team,
> 
> EKS node viewer showing only on-demand pricing , but it is not showing spot pricing. The spot instance created by karpenter and I can able to see label "karpenter.sh/capacity-type | spot" in spot node.
> 
> ![image](https://private-user-images.githubusercontent.com/144238035/278308818-f628f4e1-496f-400c-a050-8ae958a6f5cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3MTA3NjAsIm5iZiI6MTcxMDcxMDQ2MCwicGF0aCI6Ii8xNDQyMzgwMzUvMjc4MzA4ODE4LWY2MjhmNGUxLTQ5NmYtNDAwYy1hMDUwLThhZTk1OGE2ZjVjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QyMTIxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YTUzYzZhZmM0NTQ1Yzk5ZDQ2OWUxYTNhZDMwZDFlNjZjYjFiNGQwNDUzNmIwZmI3MDBiMzU1NDQ5MGViOWE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.skVwxWz965s8c6FAzjnOz00PgZ1CvM6kpMlgbwlGEdA)


I am experiencing the same issue.

**Comment by tzneal** [comment]
Yes, please check the permissions on the IAM role that you are using.

**Comment by Tr3V3rn** [comment]
Thanks the issue was resolved. I needed to correctly set the region in my
AWS config file as I am not using us-east-1 hence it was not getting the
spot price information.

**Comment by brito-rafa** [comment]
It was said before, but just reinstating: after exporting `AWS_REGION=us-east-1` variable, spot pricing has started to show up.
